### DISCUSSION TITLE
Set global password complexity

### DISCUSSION THREAD

**Discussion by Hardexit** [dani-garcia/vaultwarden#4618] [Q&A]
Hello,

Is there a way to set the default password complexity for an instance?

**Comment by jobritz** [comment]
You can set it in an organization with an organization policy, but this will only be enforced on members so I think that currently there is no way to set a default compexity for all users in an instance

**Reply by Hardexit** [reply]
But that sounds in the interface so that the user can no longer change it?
I just wanted to assign a standard thickness guideline, 30 letters, characters, numbers, special characters and so on

**Reply by jobritz** [reply]
No the users can still change their passwords, but they have to follow the guidline you set, so their password will need 30 letters, characters, numbers and special charachters.

**Reply by Hardexit** [reply]
I then see the problem that you have to be able to change this.
there are still too many forms that limit the character length or do not allow special characters.

**Reply by BlackDex** [reply]
The only other way is to change and build the web-vault your self. The default is baked into the clients and we are not able to change this at runtime.

**Comment by M1CK431** [comment]
Not a "bulletproof" solution, but it works for now and doesn’t require rebuilding anything or downgrading to an older version of the web app:
1. Locate the web app directory (in my case, it’s at `/usr/share/webapps/vaultwarden-web`).
2. `cd /usr/share/webapps/vaultwarden-web/app`
3. Run: `sed -i 's/minimumPasswordLength=12/minimumPasswordLength=8/g' main.*.js`
4. Reload the web app in your browser, log in, and change your password 🎉

Note that since this is a **client-side** limitation, a more proper approach would be to use the backend API directly (e.g., via `curl` or [vaultwarden-cli](https://github.com/haydonryan/vaultwarden-cli)). I didn’t test that method myself, as I prefer to allow shorter passwords for all users on my Vaultwarden instance anyway.

Since this modifies frontend files, **Vaultwarden updates will overwrite `main.*.js`**. If you want it to persist, you could:
- Add the `sed` command to a post-update hook or systemd service
- Use a reverse proxy to serve a custom JS file
- Or simply document that it needs to be reapplied after each update
